### PR TITLE
Set up GitHub workflow

### PR DESCRIPTION
**Pull request by ShrirajHegde**
Continuously build static binary and upload using GitHub workflow

### CODE REVIEW

**Comment by aristocratos** [comment]
Great idea!
I would however like to ask that you add an `x86_64` tag to the binary.
And if you want to, add commands for compiling for the other architectures.

Below is an excerpt from my release compile script that might be of use (for the compiler names):
```bash
#? X86_64
if [[ ! -e "packages/btop-${version}-linux-x86_64.tbz" ]]; then
        echo -e "${color}1% ${normal}Building btop++ v$version for x86_64 ${reset}"
        make distclean >/dev/null
        make CXX=g++-11 ARCH=x86_64 STATIC=true QUIET=true | tail -n3 | head -n1 || exit 1
        echo

        echo -e "${color}5% ${normal}Creating package for x86_64 ${reset}\n"
        tar -C ./release -chjf "packages/btop-${version}-linux-x86_64.tbz" ./ --transform='s,^\./,,' || exit 1
fi

#? i686
if [[ ! -e "packages/btop-${version}-linux-i686.tbz" ]]; then
        echo -e "${color}6% ${normal}Building btop++ v$version for i686 ${reset}"
        make distclean >/dev/null
        make CXX=i686-linux-gnu-g++-11 ARCH=i686 STATIC=true QUIET=true | tail -n3 | head -n1 || exit 1
        echo

        echo -e "${color}10% ${normal}Creating package for i686 ${reset}\n"
        tar -C ./release -chjf "packages/btop-${version}-linux-i686.tbz" ./ --transform='s,^\./,,' || exit 1
fi

#? AARCH64
if [[ ! -e "packages/btop-${version}-linux-aarch64.tbz" ]]; then
        echo -e "${color}11% ${normal}Building btop++ v$version for aarch64 ${reset}"
        make distclean >/dev/null
        make CXX=aarch64-linux-gnu-g++-11 ARCH=aarch64 STATIC=true QUIET=true | tail -n3 | head -n1 || exit 1
        echo

        echo -e "${color}15% ${normal}Creating package for aarch64 ${reset}\n"
        tar -C ./release -chjf "packages/btop-${version}-linux-aarch64.tbz" ./ --transform='s,^\./,,' || exit 1
fi

#? ARMEL
if [[ ! -e "packages/btop-${version}-linux-armel.tbz" ]]; then
        echo -e "${color}16% ${normal}Building btop++ v$version for armel ${reset}"
        make distclean >/dev/null
        make CXX=arm-linux-gnueabi-g++-11 ARCH=armel STATIC=true QUIET=true | tail -n3 | head -n1 || exit 1
        echo

        echo -e "${color}20% ${normal}Creating package for armel ${reset}\n"
        tar -C ./release -chjf "packages/btop-${version}-linux-armel.tbz" ./ --transform='s,^\./,,' || exit 1
fi

#? ARMHF
if [[ ! -e "packages/btop-${version}-linux-armhf.tbz" ]]; then
        echo -e "${color}21% ${normal}Building btop++ v$version for armhf ${reset}"
        make distclean >/dev/null
        make CXX=arm-linux-gnueabihf-g++-11 ARCH=armhf STATIC=true QUIET=true | tail -n3 | head -n1 || exit 1
        echo

        echo -e "${color}25% ${normal}Creating package for armhf ${reset}\n"
        tar -C ./release -chjf "packages/btop-${version}-linux-armhf.tbz" ./ --transform='s,^\./,,' || exit 1
fi

#? RISCV64
if [[ ! -e "packages/btop-${version}-linux-riscv64.tbz" ]]; then
        echo -e "${color}26% ${normal}Building btop++ v$version for riscv64 ${reset}"
        make distclean >/dev/null
        make CXX=riscv64-linux-gnu-g++-11 ARCH=riscv64 STATIC=true QUIET=true | tail -n3 | head -n1 || exit 1
        echo

        echo -e "${color}30% ${normal}Creating package for riscv64 ${reset}\n"
        tar -C ./release -chjf "packages/btop-${version}-linux-riscv64.tbz" ./ --transform='s,^\./,,' || exit 1
fi

#? SPARC64
if [[ ! -e "packages/btop-${version}-linux-sparc64.tbz" ]]; then
        echo -e "${color}31% ${normal}Building btop++ v$version for sparc64 ${reset}"
        make distclean >/dev/null
        make CXX=sparc64-linux-gnu-g++-11 ARCH=sparc64 STATIC=true QUIET=true | tail -n3 | head -n1 || exit 1
        echo

        echo -e "${color}35% ${normal}Creating package for sparc64 ${reset}\n"
        tar -C ./release -chjf "packages/btop-${version}-linux-sparc64.tbz" ./ --transform='s,^\./,,' || exit 1
fi
```

If you don't feel like doing it, no problem, great work either way :)

File: README.md [hunk 1]
```diff
@@ -12,24 +12,36 @@
...
+      - [Main UI showing details for a selected process](#main-ui-showing-details-for-a-selected-process)
+      - [Main UI in TTY mode](#main-ui-in-tty-mode)
+      - [Main UI with custom options](#main-ui-with-custom-options)
+      - [Main-menu](#main-menu)
+      - [Options-menu](#options-menu)
+      - [Help-menu](#help-menu)
+  - [Installation](#installation)
+  - [Compilation](#compilation)
+  - [Install the snap](#install-the-snap)
+      - [Command line options](#command-line-options)
```
Comment (aristocratos) [inline README.md:43]: Not sure why this is indented under "Install the snap"?
Also missing "Configurability" category.

Comment (ShrirajHegde) [reply README.md:43]: I am not sure what happened in Readme.md
I did not intend any indentation or any modification below Index

Comment (ShrirajHegde) [reply README.md]: Okay, it is the extension `markdownlint` in VS Code which made that horrible formatting.

Getting rid of that solved the problem.

**Comment by ShrirajHegde** [comment]
> Great idea! I would however like to ask that you add an `x86_64` tag to the binary. And if you want to, add commands for compiling for the other architectures.

Can you give me a list of dependencies for cross compilation? I don't have experience in cross compilation.

**Comment by aristocratos** [comment]
@ShrirajHegde
Are the steps in the workflow run in the same shell after each other? In that case you would need a `make distclean` in between.
The build tools looks fine, `gcc-11-aarch64-linux-gnu` should pull in everything needed.

It looks like `make` didn't see the `ARCH=aarch64`, which is weird. But could be some weirdness caused by not cleaning up possibly...

**Comment by ShrirajHegde** [comment]
> @ShrirajHegde Are the steps in the workflow run in the same shell after each other? In that case you would need a `make distclean` in between. The build tools looks fine, `gcc-11-aarch64-linux-gnu` should pull in everything needed.
> 
> It looks like `make` didn't see the `ARCH=aarch64`, which is weird. But could be some weirdness caused by not cleaning up possibly...

yes, it runs in the same environment, let me try distclean

**Comment by ShrirajHegde** [comment]
I think it should be working, I am not sure why the action isn't being triggered.

**Comment by aristocratos** [comment]
@ShrirajHegde 
Awesome work :)

Is it possible to split the artifacts in to separate files instead of one big?

**Comment by ShrirajHegde** [comment]
> @ShrirajHegde Awesome work :)
> 
> Is it possible to split the artifacts in to separate files instead of one big?

Yes, I can just use the upload action 7 times for each file. The only tiny inconvenience is it zips the artifact even if there's a single file.

**Comment by ShrirajHegde** [comment]
I'll try to find an alternative action that doesn't zip the artifact.

BTW since you mentioned the release script, if the script is added as a workflow, you can just create a version tag and the workflow can build and upload to releases. Will add that in a separate PR.

**Comment by aristocratos** [comment]
@ShrirajHegde 
> I'll try to find an alternative action that doesn't zip the artifact.

I don't mind it being zipped, (compression = bandwidth savings), as long as they aren't in one big zip.

> BTW since you mentioned the release script, if the script is added as a workflow, you can just create a version tag and the workflow can build and upload to releases. Will add that in a separate PR.

Thanks, but there's no need really. I need to compile locally for all platforms to make sure there are no errors before creating a release. And it only takes about a minute.

**Comment by aristocratos** [comment]
@ShrirajHegde 
Looks to be working great, thanks for the contribution/workflow magic :)

**Pull request merged**